[Regression] net8.0 builds are using Microsoft.iOS 17.2 even when built with net8.0-ios17.0

### Current behavior

Some canaries fail:

> ##[error]CSC(0,0): Error CS1705: Assembly 'Uno.UI' with identity 'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.iOS, Version=17.2.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065' which has a higher version than referenced assembly 'Microsoft.iOS' with identity 'Microsoft.iOS, Version=17.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'


It looks like the .NET 8 SDK doesn't work well with target platform versions?

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/build/ci/.azure-devops-package-netcoremobile.yml
+++ b/build/ci/.azure-devops-package-netcoremobile.yml
@@ -49,7 +49,9 @@ jobs:
 
   - template: templates/gitversion.yml
 
-  - template: templates/dotnet-mobile-install-windows.yml
+  # Building with .NET 9 SDK for net8.0-ios17.0 to work.
+  # It looks like building net8.0-ios17.0 with .NET 8 SDK will end up using Microsoft.iOS 17.2 assembly.
+  - template: templates/dotnet9-mobile-install-windows.yml
 
   # This task is required to run in separately to avoid hitting targets file
   # caching issues with msbuild 17.3 or later

--- a/build/ci/templates/dotnet9-mobile-install-windows.yml
+++ b/build/ci/templates/dotnet9-mobile-install-windows.yml
@@ -1,0 +1,25 @@
+steps:
+
+  - bash: |
+      rm global.json
+      cp build/ci/net9/global.json global.json
+    displayName: "Using .NET 9 global.json"
+
+  - task: UseDotNet@2
+    displayName: 'Use .NET SDK'
+    retryCountOnTaskFailure: 3
+    inputs:
+      packageType: sdk
+      useGlobalJson: true
+      includePreviewVersions: true
+
+
+  - template: jdk-setup.yml
+
+  - powershell: |
+      & dotnet tool update --global uno.check --version 1.22.0 --add-source https://api.nuget.org/v3/index.json
+      & uno-check -v --ci --non-interactive --fix --skip androidsdk --skip androidemulator --skip xcode --skip gtk3 --skip vswin --skip vsmac --manifest https://raw.githubusercontent.com/unoplatform/uno.check/f2ba48c5cb67d59d8025ca96e436e1a9f6f3012c/manifests/uno.ui-preview-major.manifest.json
+    displayName: Install .NET Workloads
+    errorActionPreference: continue
+    ignoreLASTEXITCODE: true
+    retryCountOnTaskFailure: 3

